### PR TITLE
chore(argo-cd): Remove duplicate ApplicationSet features

### PR DESCRIPTION
**Pull request by pdrastil**
Removed features that can be automatically detected or have same functionality as already present feature.

---

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/templates/argocd-applicationset/role.yaml [hunk 1]
```diff
@@ -48,40 +48,31 @@ rules:
...
     - get
     - list
     - watch
-  # Leader election
   - apiGroups:
     - ""
+    - coordination.k8s.io
     resources:
     - configmaps
+    - leases
```
Comment (pdrastil) [inline charts/argo-cd/templates/argocd-applicationset/role.yaml:60]: Removed extra section as both configmaps and leases have same rbac.

Comment (mkilchhofer) [reply charts/argo-cd/templates/argocd-applicationset/role.yaml:60]: I dont think think this is a good idea.
Why should they be the same (esp. the verbs)?

Comment (pdrastil) [reply charts/argo-cd/templates/argocd-applicationset/role.yaml:60]: Check the verbs for configmaps and leases when you put them side-by-side.

Comment (pdrastil) [reply charts/argo-cd/templates/argocd-applicationset/role.yaml]: I've kept it separated even when verbs are the same. Just joined configmap groups.

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**